src/DataCollection.UWP/MainPage.xaml

```diff
@@ -0,0 +1,76 @@
+﻿<Page
+    x:Class="Esri.ArcGISRuntime.ExampleApps.DataCollection.UWP.MainPage"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:esri="using:Esri.ArcGISRuntime.UI.Controls"
+    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+    xmlns:toolkit="using:Esri.ArcGISRuntime.Toolkit.UI.Controls"
+    xmlns:utils="using:Esri.ArcGISRuntime.ExampleApps.DataCollection.Shared.Utils"
+    xmlns:converters="using:Esri.ArcGISRuntime.ExampleApps.DataCollection.Shared.Converters"
+    RequestedTheme="Dark"
+    mc:Ignorable="d">
+    
+    <Page.Resources>
+        <ResourceDictionary>
+            <converters:BoolToVisibilityConverter x:Key="BoolToVisibilityConverter"/>
+            <converters:NullToVisibilityConverter x:Key="NullToVisibilityConverter"/>
+        </ResourceDictionary>
+    </Page.Resources>
+    
+    <Grid Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
+        <Grid.RowDefinitions>
+            <RowDefinition Height="Auto"/>
+            <RowDefinition Height="*"/>
+        </Grid.RowDefinitions>
+        <Grid.ColumnDefinitions>
+            <ColumnDefinition Width="Auto" MinWidth="300" MaxWidth="300"/>
+            <ColumnDefinition Width="Auto" MinWidth="300" MaxWidth="300"/>
+            <ColumnDefinition Width="*"/>
+        </Grid.ColumnDefinitions>
+        <esri:MapView x:Name="MapView" Map="{x:Bind MainViewModel.MapViewModel.Map, Mode=OneWay}" 
+                      Grid.Row="1" Grid.RowSpan="2" Grid.Column="0" Grid.ColumnSpan="3"
+                      utils:MapViewExtensions.IdentifyController="{x:Bind MainViewModel.IdentifyController, Mode=OneWay}">
+            <utils:MapViewExtensions.ViewpointController>
+                <utils:ViewpointController  Viewpoint="{x:Bind MainViewModel.MapViewModel.AreaOfInterest, Mode=OneWay}"/>
+            </utils:MapViewExtensions.ViewpointController>
+            <utils:MapViewExtensions.LocationDisplayController>
+                <utils:LocationDisplayController DataSource="{x:Bind MainViewModel.MapViewModel.LocationDataSource, Mode=OneWay}"/>
+            </utils:MapViewExtensions.LocationDisplayController>
+        </esri:MapView>
+
+        <!-- Compass control that displays when user rotates the map -->
+        <toolkit:Compass GeoView="{Binding ElementName=MapView, Mode=OneWay}" AutoHide="True"
+                      Margin="20"  Grid.Column="2" Grid.Row="1" />
+
+        <!-- Title bar with top buttons-->
+        <CommandBar Grid.Row="0" Grid.ColumnSpan="3">
+            <CommandBar.Content>
+                <TextBlock Text="{x:Bind MainViewModel.MapViewModel.Map.Item.Title, Mode=OneWay}"  Foreground="White" FontSize="25" FontWeight="SemiBold"
```

Comment:
Only other change needed for fully dynamic themes is removing the explicit foreground color. I tested and the theme picks the right color consistently.